batch/batch/driver/job.py

```diff
@@ -364,9 +364,13 @@ async def job_config(app, record, attempt_id):
         name = service_account['name']
 
         sa = await k8s_cache.read_service_account(name, namespace)
-        assert len(sa.secrets) == 1
-
-        token_secret_name = sa.secrets[0].name
+        if sa.secrets is not None:
+            # ServiceAccounts created prior to Kubernetes 1.24 have autogenerated secrets
+            assert len(sa.secrets) == 1
+            token_secret_name = sa.secrets[0].name
+        else:
+            # ServiceAccounts post v1.24 don't have autogenerated secrets and we make those ourselves
+            token_secret_name = f'{name}-token'
```

Comment:
Hmm. So, you didn't change auth's create user stuff which makes me think that this stuff is only used by CI and only the service accounts you added below are relevant. Is that right?

Reply:
Yes, this only pertains to k8s service accounts not GSAs or Azure SPs